tests/serverpod_test_server/test_integration/test_tools/authentication_validation_test.dart

```diff
@@ -0,0 +1,98 @@
+import 'package:serverpod/serverpod.dart';
+import 'package:test/test.dart';
+
+import 'serverpod_test_tools.dart';
+
+void main() {
+  group('Given AuthenticatedTestToolsEndpoint', () {
+    withServerpod(
+      (endpoints, session) {
+        test(
+            'when not authenticated and calling returnsString then throws UnauthenticatedEndpointCallTestException',
+            () async {
+          session = await session.copyWith(getAuthenticationInfo: () => null);
+
+          final result =
+              endpoints.authenticatedTestTools.returnsString(session, "Hello");
+          await expectLater(
+              result, throwsA(isA<UnauthenticatedEndpointCallTestException>()));
```

Comment:
Since we're stripping out the transport layer, we basically have to decide wether we'd like to either expose internal exceptions from Serverpod or create test domain specific ones. I opted for the latter here, since it seems more prudent to have a separation and avoid user dependencies on internal Serverpod Exceptions.

These two exceptions are mapping 1:1 to internal Serverpod exceptions (see `packages/serverpod_test/lib/src/function_call_wrappers.dart`) so their only function is to create a layer of indirection.